File: Client/java/apisdk/list/ListRequestBuilder.java
package apisdk.list;

import apisdk.list.chromosomes.ChromosomesRequestBuilder;
import apisdk.list.files.FilesRequestBuilder;
import apisdk.list.hubgenomes.HubGenomesRequestBuilder;
import apisdk.list.publichubs.PublicHubsRequestBuilder;
import apisdk.list.schema.SchemaRequestBuilder;
import apisdk.list.tracks.TracksRequestBuilder;
import apisdk.list.ucscgenomes.UcscGenomesRequestBuilder;
import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.RequestAdapter;
import java.util.HashMap;
import java.util.Objects;
/**
 * Builds and executes requests for operations under /list
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class ListRequestBuilder extends BaseRequestBuilder {
    /**
     * The chromosomes property
     * @return a {@link ChromosomesRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public ChromosomesRequestBuilder chromosomes() {
        return new ChromosomesRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The files property
     * @return a {@link FilesRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public FilesRequestBuilder files() {
        return new FilesRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The hubGenomes property
     * @return a {@link HubGenomesRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public HubGenomesRequestBuilder hubGenomes() {
        return new HubGenomesRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The publicHubs property
     * @return a {@link PublicHubsRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public PublicHubsRequestBuilder publicHubs() {
        return new PublicHubsRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The schema property
     * @return a {@link SchemaRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public SchemaRequestBuilder schema() {
        return new SchemaRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The tracks property
     * @return a {@link TracksRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public TracksRequestBuilder tracks() {
        return new TracksRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The ucscGenomes property
     * @return a {@link UcscGenomesRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public UcscGenomesRequestBuilder ucscGenomes() {
        return new UcscGenomesRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * Instantiates a new {@link ListRequestBuilder} and sets the default values.
     * @param pathParameters Path parameters for the request
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public ListRequestBuilder(@jakarta.annotation.Nonnull final HashMap<String, Object> pathParameters, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/list", pathParameters);
    }
    /**
     * Instantiates a new {@link ListRequestBuilder} and sets the default values.
     * @param rawUrl The raw URL to use for the request builder.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public ListRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/list", rawUrl);
    }
}

File: Client/php/EscapedList/ListRequestBuilder.php
<?php

namespace ApiSdk\EscapedList;

use ApiSdk\EscapedList\Chromosomes\ChromosomesRequestBuilder;
use ApiSdk\EscapedList\Files\FilesRequestBuilder;
use ApiSdk\EscapedList\HubGenomes\HubGenomesRequestBuilder;
use ApiSdk\EscapedList\PublicHubs\PublicHubsRequestBuilder;
use ApiSdk\EscapedList\Schema\SchemaRequestBuilder;
use ApiSdk\EscapedList\Tracks\TracksRequestBuilder;
use ApiSdk\EscapedList\UcscGenomes\UcscGenomesRequestBuilder;
use Microsoft\Kiota\Abstractions\BaseRequestBuilder;
use Microsoft\Kiota\Abstractions\RequestAdapter;

/**
 * Builds and executes requests for operations under /list
*/
class ListRequestBuilder extends BaseRequestBuilder 
{
    /**
     * The chromosomes property
    */
    public function chromosomes(): ChromosomesRequestBuilder {
        return new ChromosomesRequestBuilder($this->pathParameters, $this->requestAdapter);
    }
    
    /**
     * The files property
    */
    public function files(): FilesRequestBuilder {
        return new FilesRequestBuilder($this->pathParameters, $this->requestAdapter);
    }
    
    /**
     * The hubGenomes property
    */
    public function hubGenomes(): HubGenomesRequestBuilder {
        return new HubGenomesRequestBuilder($this->pathParameters, $this->requestAdapter);
    }
    
    /**
     * The publicHubs property
    */
    public function publicHubs(): PublicHubsRequestBuilder {
        return new PublicHubsRequestBuilder($this->pathParameters, $this->requestAdapter);
    }
    
    /**
     * The schema property
    */
    public function schema(): SchemaRequestBuilder {
        return new SchemaRequestBuilder($this->pathParameters, $this->requestAdapter);
    }
    
    /**
     * The tracks property
    */
    public function tracks(): TracksRequestBuilder {
        return new TracksRequestBuilder($this->pathParameters, $this->requestAdapter);
    }
    
    /**
     * The ucscGenomes property
    */
    public function ucscGenomes(): UcscGenomesRequestBuilder {
        return new UcscGenomesRequestBuilder($this->pathParameters, $this->requestAdapter);
    }
    
    /**
     * Instantiates a new ListRequestBuilder and sets the default values.
     * @param array<string, mixed>|string $pathParametersOrRawUrl Path parameters for the request or a String representing the raw URL.
     * @param RequestAdapter $requestAdapter The request adapter to use to execute the requests.
    */
    public function __construct($pathParametersOrRawUrl, RequestAdapter $requestAdapter) {
        parent::__construct($requestAdapter, [], '{+baseurl}/list');
        if (is_array($pathParametersOrRawUrl)) {
            $this->pathParameters = $pathParametersOrRawUrl;
        } else {
            $this->pathParameters = ['request-raw-url' => $pathParametersOrRawUrl];
        }
    }

}
